applications/RomApplication/python_scripts/structural_mechanics_analysis_rom.py

```diff
@@ -32,8 +32,8 @@ def ModifyInitialGeometry(self):
             data=json.load(f)
             counter = 0
             rom_dofs=self.project_parameters["solver_settings"]["rom_settings"]["number_of_rom_dofs"].GetInt()
-            Dimensions = len(data['1.0'])  #Checking for the number of DOFs (a vector or a matrix is imported to each node)
```

Comment:
This assumes that there is a node with an ID "1.0". This might not be correct in general. A general model, that for instance, is a subset of a different model might have node ID's starting from a random number

Reply:
Got it! you are right